addons/mail/models/mail_thread.py

```diff
@@ -1966,14 +1966,14 @@ def _process_attachments_for_post(self, attachments, attachment_ids, message_val
         m2m_attachment_ids = []
         if attachment_ids:
             # taking advantage of cache looks better in this case, to check
-            filtered_attachment_ids = self.env['ir.attachment'].sudo().browse(attachment_ids).filtered(
-                lambda a: a.res_model == 'mail.compose.message' and a.create_uid.id == self._uid)
+            user_attachments = self.env['ir.attachment'].sudo().browse(attachment_ids).filtered(
+                lambda a: a.create_uid.id == self._uid)
             # update filtered (pending) attachments to link them to the proper record
-            if filtered_attachment_ids:
-                filtered_attachment_ids.write({'res_model': model, 'res_id': res_id})
+            if filtered_attachments := user_attachments.filtered(lambda a: a.res_model == 'mail.compose.message'):
+                filtered_attachments.write({'res_model': model, 'res_id': res_id})
             # prevent public and portal users from using attachments that are not theirs
             if not self.env.user._is_internal():
-                attachment_ids = filtered_attachment_ids.ids
+                attachment_ids = user_attachments.ids
```

Comment:
I don't think the fix is correct:

Only self-attachments that had `res_model == 'mail.composer.message'` were returned (and relinked properly to correct `res_id` and `res_model`)

I think this comes from file upload defaulting to that state, and it's then relinked on message post. Somehow this is not the case for guests? Why is it the case?

If we remove it, then all attachments created by the user are returned.

So imagine I posted 100 attachments in the past. After this fix, if I post a new message, does it considers that my newly posted message has these 100 attachments? At a glance that's what the code does, I think

Reply:
file upload from `mail.guest` should also put the `res_model` to `mail.composer.message`, so that we can detect them as recent uploads and link them to the message being posted.

Reply:
Indeed the new posted message in this code has just its attachment and `mail.partner` doesn't send the messages with `res_mode`: `mail.composer.message`. It is actually has `res_model` as `mail.channel`.